### PR TITLE
XSI-915: Improve performance of LVHDoHBA operations by

### PR DESCRIPTION
**Pull request by BenSimsCitrix**
This commit increases performance by.

       1) Reducing expensive calls to _pathfresh
	2) Use multipath with option "-l" and then falling back to using the more expensive "-ll"
	   only if first operation it fails.
	3) Add missing vdi_epoch operations to vdi operation list

This has been through a BST.

### CODE REVIEW

File: drivers/LVHDoHBASR.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def create(self, sr_uuid, size):
         self.hbasr.attach(sr_uuid)
         if self.mpath == "true":
             self.mpathmodule.refresh(self.SCSIid, 0)
-        self._pathrefresh(LVHDoHBASR)
+        self.load(sr_uuid)
```
Comment (MarkSymsCtx) [inline drivers/LVHDoHBASR.py:117]: Not required. This is called from the superclass __init__

File: drivers/LVHDoHBASR.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def attach(self, sr_uuid):
             for file in os.listdir(path):
                 self.block_setscheduler('%s/%s' % (path, file))
 
-        self._pathrefresh(LVHDoHBASR)
+        self.load(sr_uuid)
```
Comment (MarkSymsCtx) [inline drivers/LVHDoHBASR.py:135]: Ditto

File: drivers/LVHDoHBASR.py [hunk 3]
```diff
@@ -150,7 +150,7 @@ def scan(self, sr_uuid):
             if not os.path.exists(self.dconf['device']):
                 util.SMlog("@@@@@ path does not exists")
                 self.mpathmodule.refresh(self.SCSIid, 0)
-                self._pathrefresh(LVHDoHBASR)
+                self.load(sr_uuid)
```
Comment (MarkSymsCtx) [inline drivers/LVHDoHBASR.py:153]: Ditto

**Review by MarkSymsCtx — CHANGES_REQUESTED** [review]
Remove the extraneous calls to load

**Comment by BenSimsCitrix** [comment]
This has got through another BST.

**Comment by BenSimsCitrix** [comment]
Pesky "iter" bug got me again this does manifest locally when i run the tests hence why i keep getting bitten, Tests are passing now

File: tests/test_SR.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    def tearDown(self):
+        pass
+
+    def create_SR(self, cmd, dconf):
+        srcmd = mock.Mock()
+        srcmd.dconf = dconf
+        srcmd.params = {'command': cmd}
+        return SR.SR(srcmd, "some SR UUID")
+
+    def test_checkroot_no_device(self):
```
Comment (MarkSymsCtx) [inline tests/test_SR.py:22]: Do these tests need to be asserting anything after the calls to check they functioned correctly?

**Review by MarkSymsCtx — APPROVED** [review]

**Review by TimSmithCtx — APPROVED** [review]

**Pull request merged**